localstack-core/localstack/services/apigateway/next_gen/execute_api/handlers/cors.py

```diff
@@ -0,0 +1,49 @@
+import logging
+from http import HTTPMethod
+
+from localstack import config
+from localstack.aws.handlers.cors import CorsEnforcer
+from localstack.aws.handlers.cors import CorsResponseEnricher as GlobalCorsResponseEnricher
+from localstack.http import Response
+
+from ..api import RestApiGatewayHandler, RestApiGatewayHandlerChain
+from ..context import RestApiInvocationContext
+from ..gateway_response import MissingAuthTokenError
+
+LOG = logging.getLogger(__name__)
+
+
+class CorsResponseEnricher(RestApiGatewayHandler):
+    def __call__(
+        self,
+        chain: RestApiGatewayHandlerChain,
+        context: RestApiInvocationContext,
+        response: Response,
+    ):
+        """
+        This is a LocalStack only handler, to allow users to override API Gateway CORS configuration and just use the
+        default LocalStack configuration instead, to ease the usage and reduce production code changes.
+        """
+        if not config.DISABLE_CUSTOM_CORS_APIGATEWAY:
+            return
+
+        if not context.invocation_request:
+            return
```

Comment:
Question: What is the scenario in which this handler gets called without an invocation request?

Reply:
This is a very good question 😅 I've taken it from the HTTP CORS handler, probably if we encounter an unexpected exception in the `InvocationRequestParser`, we could maybe raise some kind of exception, and the response handlers are always executed, so we would avoid a weird attribute error, or trying to access `["headers"]` of `None` maybe? I think it's pretty unlikely, but better safe than sorry with the response handlers 😬